### PR TITLE
SONARJAVA-3864 Missing arguments in Deprecated annotation should be reported in its own rule

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**

### CODE REVIEW

**Comment by quentin-jaquier-sonarsource** [comment]
The first commit is about refactoring the current rules related to Deprecated annotations. The class "AbstractDeprecatedChecker" contains only static helper methods, it was in fact more a helper rather than an abstract class.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6355_java.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<p>FIXME, see https://github.com/SonarSource/rspec/pull/290</p>
```
Comment (quentin-jaquier-sonarsource) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6355_java.html:1]: When the review of this PR will be done, we should review and merge the other PR first (https://github.com/SonarSource/rspec/pull/290), and then update the metadata.

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM! I would only change the data structure in the abstract checker but I leave this to your discretion ;)

File: java-checks/src/main/java/org/sonar/java/checks/AbstractMissingDeprecatedChecker.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import static org.sonar.java.checks.helpers.DeprecatedCheckerHelper.deprecatedAnnotation;
+import static org.sonar.java.checks.helpers.DeprecatedCheckerHelper.hasJavadocDeprecatedTag;
+import static org.sonar.java.model.JUtils.isLocalVariable;
+
+public abstract class AbstractMissingDeprecatedChecker extends IssuableSubscriptionVisitor {
+
+  private static final Kind[] CLASS_KINDS = PublicApiChecker.classKinds();
+
+  private final Deque<Boolean> classOrInterfaceIsDeprecated = new LinkedList<>();
```
[the remark this answers is not in the document]
Comment (quentin-jaquier-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/AbstractMissingDeprecatedChecker.java]: Make sense to me!

**Pull request merged**